src/engine/sidechain/engineshoutcast.cpp

```diff
@@ -455,6 +496,45 @@ bool EngineShoutcast::serverConnect() {
 
 void EngineShoutcast::write(unsigned char *header, unsigned char *body,
                             int headerLen, int bodyLen) {
+    struct shoutcastCacheObject *l_SCacheObj = (struct shoutcastCacheObject *) malloc(sizeof(struct shoutcastCacheObject));
+
+    if (l_SCacheObj == NULL)
+    {
+       qDebug() << "EngineShoutcast::write: Can't allocate memory!";
+       return;
+    }
+
+    memset(l_SCacheObj, 0x00, sizeof(struct shoutcastCacheObject));
+    l_SCacheObj->header = (unsigned char *)malloc(headerLen);
+
+    if (l_SCacheObj->header == NULL) {
+       qDebug() << "EngineShoutcast::write: Can't allocate memory!";
+       free(l_SCacheObj);
+       return;
+    }
+
+    l_SCacheObj->body = (unsigned char *)malloc(bodyLen);
+
+    if (l_SCacheObj->body == NULL) {
+       qDebug() << "EngineShoutcast::write: Can't allocate memory!";
+       free(l_SCacheObj->header);
+       free(l_SCacheObj);
+       return;
+    }
+
+    memcpy(l_SCacheObj->header, header, headerLen);
+    memcpy(l_SCacheObj->body, body, bodyLen);
+
+    l_SCacheObj->headerLen = headerLen;
+    l_SCacheObj->bodyLen = bodyLen;
+
+    m_SMutex.lock();
+    m_pShoutcastCache.append(l_SCacheObj);
```

Comment:
Using the lock free https://github.com/mixxxdj/mixxx/blob/master/src/util/fifo.h
will avoid the locking. 
But be aware, it is based on a plain C fifo.